app/lib/features/events/widgets/event_date_widget.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:acter/common/utils/utils.dart';
+import 'package:acter/features/events/event_utils/event_utils.dart';
+import 'package:acter/features/events/providers/event_providers.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter/material.dart';
+
+class EventDateWidget extends StatelessWidget {
+  final CalendarEvent calendarEvent;
+  final double size;
+
+  const EventDateWidget({
+    super.key,
+    required this.calendarEvent,
+    this.size = 70,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return _buildEventDate(context);
+  }
+
+  Widget _buildEventDate(BuildContext context) {
+    final day = getDayFromDate(calendarEvent.utcStart());
+    final month = getMonthFromDate(calendarEvent.utcStart());
+
+    return Card(
+      color: getColorBasedOnEventType(context),
+      child: Container(
+        height: size,
+        width: size,
+        alignment: Alignment.center,
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: [
+            Text(month),
+            Text(day, style: Theme.of(context).textTheme.titleLarge),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Color getColorBasedOnEventType(BuildContext context) {
+    if (getEventType(calendarEvent) == EventFilters.past) {
```

Comment:
I am seeing that `getEventType` calculation rather often. It is not that large but we might want to consider putting it into a `eventTypeForEvent(eventId)` provider instead at some point...

Reply:
Agree but prefer to do in separate PR so that this big PR not get stretched out much.